internal/controller/clientprofilemapping_controller.go

```diff
@@ -18,45 +18,184 @@ package controller
+//+kubebuilder:rbac:groups=csi.ceph.io,resources=clientprofilemappings/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=csi.ceph.io,resources=clientprofilemappings/finalizers,verbs=update
+//+kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch;create;update;delete
+
 // ClientProfileMappingReconciler reconciles a ClientProfileMapping object
 type ClientProfileMappingReconciler struct {
 	client.Client
 	Scheme *runtime.Scheme
 }
 
-//+kubebuilder:rbac:groups=csi.ceph.io,resources=clientprofilemappings,verbs=get;list;watch;create;update;patch;delete
-//+kubebuilder:rbac:groups=csi.ceph.io,resources=clientprofilemappings/status,verbs=get;update;patch
-//+kubebuilder:rbac:groups=csi.ceph.io,resources=clientprofilemappings/finalizers,verbs=update
-
-// Reconcile is part of the main kubernetes reconciliation loop which aims to
-// move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the ClientProfileMapping object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.17.3/pkg/reconcile
-func (r *ClientProfileMappingReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+// A local reconcile object tied to a single reconcile iteration
+type ClientProfileMappingReconcile struct {
+	ClientProfileMappingReconciler
 
-	// TODO(user): your logic here
+	ctx                      context.Context
+	log                      logr.Logger
+	req                      ctrl.Request
+	clientProfileMappingList csiv1a1.ClientProfileMappingList
+}
 
-	return ctrl.Result{}, nil
+// csiClusterMappingRecord represnet the structure to serialize a csi mapping
+// record in Ceph CSI's config, configmap
+type csiClusterMappingRecord struct {
+	ClusterIdMapping map[string]string   `json:"clusterIdMapping,omitempty"`
+	RbdPoolIdMapping []map[string]string `json:"RBDPoolIDMapping,omitempty"`
 }
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *ClientProfileMappingReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	genChangedPredicate := predicate.GenerationChangedPredicate{}
+
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&csiv1alpha1.ClientProfileMapping{}).
+		For(&csiv1a1.ClientProfileMapping{}).
+		// TODO: This watch is probalmatic, it will trigger a reconicle for each
+		// mapping resource that exists in the configmap namespace whenever the configmap
+		// is changed. This is unnececary as this reconile build its desired state based
+		// on all mapping resources withing a single reconicle
+		Owns(
+			&corev1.ConfigMap{},
+			builder.MatchEveryOwner,
+			builder.WithPredicates(
+				utils.NamePredicate(utils.CsiConfigVolume.Name),
+				genChangedPredicate,
```

Comment:
generation on configmap doesn't seem usual.

Reply:
But it is supported.

Reply:
ack, if it does, can be resolved.